client/landing/stepper/declarative-flow/internals/steps-repository/woo-confirm/index.tsx

```diff
@@ -64,49 +52,44 @@ const WooConfirm: Step = function WooCommerceConfirm( { navigation } ) {
 		requestLatestAtomicTransfer( siteId );
 	}, [ requestLatestAtomicTransfer, siteId ] );
 
-	const eligibilityHolds = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityHolds( siteId ),
-		[ siteId ]
-	);
-	const eligibilityWarnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityWarnings( siteId ),
-		[ siteId ]
-	);
-	const wpcomSubdomainWarning = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getWpcomSubdomainWarning( siteId ),
-		[ siteId ]
-	);
-	const warnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getNonSubdomainWarnings( siteId ),
-		[ siteId ]
-	);
-	const latestAtomicTransfer = useSelect(
-		( select ) => ( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransfer( siteId || 0 ),
-		[ siteId ]
-	);
-	const latestAtomicTransferError = useSelect(
-		( select ) =>
-			( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransferError( siteId || 0 ),
-		[ siteId ]
-	);
-	const productsList = useSelect(
-		( select ) => ( select( PRODUCTS_LIST_STORE ) as ProductsListSelect ).getProductsList(),
-		[]
-	);
-	const requiresUpgrade = useSelect(
-		( select ) => ( select( SITE_STORE ) as SiteSelect ).requiresUpgrade( siteId ),
+	const {
+		eligibilityHolds,
+		eligibilityWarnings,
+		wpcomSubdomainWarning,
+		warnings,
+		latestAtomicTransfer,
+		latestAtomicTransferError,
+		productsList,
+		requiresUpgrade,
+		isAtomicSite,
+		stepProgress,
+	} = useSelect(
+		( select ) => {
+			const {
+				getEligibilityHolds,
+				getEligibilityWarnings,
+				getWpcomSubdomainWarning,
+				getNonSubdomainWarnings,
+			} = select( AutomatedTransferEligibility.store );
+			const {
+				getSiteLatestAtomicTransfer,
+				getSiteLatestAtomicTransferError,
+				requiresUpgrade,
+				isSiteAtomic,
+			} = select( SITE_STORE ) as SiteSelect;
```

Comment:
I'd personally select from separate stores in separate `useSelect` calls. Especially when they don't depend on each other, like result of one `select` being an input to another.

It breaks a big blob of code into several smaller ones, and also avoids calling selectors for store A when there's and update only in store B.

Reply:
This does bring a couple of `@wordpress/data` potential improvement questions, though: 

* Should we do some work in `@wordpress/data` to optimize for both scenarios?
* If the answer to the first question is "no", then should we update the docs to suggest one store per one `useSelect()` call for optimal performance?

Reply:
Cool, that's good to know.